scheduling/tasks.go

```diff
@@ -70,50 +77,61 @@ func (t *TaskMetadata) Unmarshal(data []byte) error {
 
 	var temp tempConfig
 	if err := json.Unmarshal(data, &temp); err != nil {
-		return fmt.Errorf("failed to deserialize task metadata: %w", err)
+		errMessage := fmt.Errorf("failed to deserialize task metadata: %w", err)
+		return fferr.NewInternalError(errMessage)
 	}
 	if temp.ID == 0 {
-		return fmt.Errorf("task metadata is missing ID")
+		return fferr.NewInvalidArgumentError(fmt.Errorf("task metadata is missing ID"))
 	}
 	t.ID = temp.ID
 
 	if temp.Name == "" {
-		return fmt.Errorf("task metadata is missing Name")
+		return fferr.NewInvalidArgumentError(fmt.Errorf("task metadata is missing name"))
 	}
 	t.Name = temp.Name
 
-	if temp.TaskType == "" || (temp.TaskType != ResourceCreation && temp.TaskType != HealthCheck && temp.TaskType != Monitoring) {
-		return fmt.Errorf("unknown task type: %s", temp.TaskType)
+	if temp.TaskType == "" {
+		return fferr.NewInvalidArgumentError(fmt.Errorf("task metadata is missing TaskType"))
+	}
+	if temp.TaskType != ResourceCreation || temp.TaskType != HealthCheck || temp.TaskType != Monitoring {
```

Comment:
this always evaluates to true

Reply:
i think you want something like this

```
func isValidTaskType(tType TaskType) bool {
    switch tType {
    case ResourceCreation, HealthCheck, Monitoring:
        return true
    default:
        return false
    }
}
```

Reply:
goland points these little things out for me. Does vscode not do it? If not you should consider switching lol